iaia/magicmodule.py

```diff
@@ -56,27 +59,45 @@ def __repr__(self):
     def __call__(self, *args, **kw):
         key = self.call_key(*args, **kw)
         if key not in self.funcs:
-            self.make_function(*args, **kw)
+            self.make_functions(*args, **kw)
         exc = None
-        try:
-            return self.funcs[key](*args, **kw)
-        except Exception as e:
-            exc = e
-        print(f"Attempting to fix exception {exc}...")
-        self.fix_function(exc, *args, **kw)
-        return self.funcs[key](*args, **kw)
+        for func_i, func in self.funcs[key].items():
+            try:
+                return func(*args, **kw)
+            except Exception as e:
+                exc = e
+            print(f"Attempting to fix exception {exc} in func {func_i}...")
+            func = self.fix_function(func_i, exc, *args, **kw)
+            return func(*args, **kw)
```

Comment:
I don't see how any of the other function definitions could be called... won't the first either return or error out?

I'm also unsure why `self.funcs[key]` is a dict instead of a list? It'll always be `{0: func_0, 1: func_1, ...}` won't it?

Reply:
Good catch on point 1, wow it's been a while and I'm rusty! Fixing both